File: Exercises/Results/toll8030/unit_test.vb
Module unit_test
    Function test_user() As Boolean
        Dim t As Boolean = True

        t = test_create()

        If t Then
            MsgBox("ok")
        Else
            MsgBox("not ok")
        End If
        Return t
    End Function

    Function test_create() As Boolean
        Dim t As Boolean = False
        Dim User As Object = User().ToString
        If User IsNot Nothing Then
            t = True
        End If
        Return t
    End Function

    Sub User()
        Dim first, last, email As String
        first = ""
        last = ""
        email = ""
        Dim name As Array
        name = {first, last}


    End Sub

    Sub UserList(user)
        Dim list As ArrayList
        list.Add(user)

    End Sub

    Function test_name() As Boolean
        Dim t As Boolean = False
        If User().name IsNot Nothing Then
            t = True
        End If
        Return t
    End Function
    Function test_email() As Boolean
        Dim t As Boolean = False
        If User().email IsNot Nothing Then
            t = True
        End If
        Return t
    End Function

    Sub Main()
        test_user()
    End Sub

End Module

File: Exercises/Results/quin9856/unit_test.py


def test_user():
    test_first(set_users()[0].first)
    test_last(set_users()[1].last)
    test_email(set_users()[2].email)

    names = [['tom', 'sawyer', 'saw@email.com'], ['ron', 'dog', 'dog@email.com'], ['kay', 'johnson', 'kay@email.com']]
    add_users(names)

def test_create():
    try:
        user = User("", "", "")
        return user
    except:
        print('object was not created')

class User(object):
    def __init__(self, first, last, email):
        self.first = first
        self.last = last
        self.email = email

#Sets 4 users and puts in a list
def set_users():
    user1 = User('Jon', 'Doe', 'doe@email.com')
    user2 = User('kelly', 'boo', 'boo@email.com')
    user3 = User('jack', 'stacks', 'stacks@email.com')
    user4 = User('don', 'baller', 'baller@email.com')
    userList = [user1, user2, user3, user4]
    return userList

def test_first(userFirst):
    try:
        if userFirst != "":
            pass
    except:
        print("No first name found")

def test_last(userLast):
    try:
        if userLast != "":
            pass
    except:
        print("No first name found")

def test_email(userEmail):
    try:
        if userEmail != "":
            pass
    except:
        print("No first name found")


def add_users(list):
    new_list = []
    for x in list:
        user = User(x[0],x[1],x[2])
        new_list.append(user)
    return new_list

try:
    test_user()
    print("It worked")
except:
    print("something failed")


